public/app/plugins/datasource/graphite/datasource_integration.test.ts

```diff
@@ -0,0 +1,154 @@
+import { GraphiteDatasource } from './datasource';
+import { TemplateSrv } from 'app/features/templating/template_srv';
+import { BackendSrv } from 'app/core/services/backend_srv';
+import { of } from 'rxjs';
+import { ContextSrv, User } from '../../../core/services/context_srv';
+import { setBackendSrv } from '@grafana/runtime';
+
+jest.mock('@grafana/runtime', () => ({
+  ...(jest.requireActual('@grafana/runtime') as unknown as object),
+}));
+
+interface Context {
+  templateSrv: TemplateSrv;
+  ds: GraphiteDatasource;
+}
+
+describe('graphiteDatasource integration with backendSrv and fetch', () => {
+  let ctx = {} as Context;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    const instanceSettings = {
+      url: '/api/datasources/proxy/1',
+      name: 'graphiteProd',
+      jsonData: {
+        rollupIndicatorEnabled: true,
+      },
+    };
+    const templateSrv = new TemplateSrv();
+    const ds = new GraphiteDatasource(instanceSettings, templateSrv);
+    ctx = { templateSrv, ds };
```

Comment:
`TemplateSrv` is created as a default value in the constructor, and it doesn't seem to be used in the test so you can also simplify it to:

```suggestion
    const ds = new GraphiteDatasource(instanceSettings);
    ctx = { ds };
```

Reply:
Can do!